ui/perfherder/graphs/GraphsContainer.jsx

```diff
@@ -332,6 +344,25 @@ class GraphsContainer extends React.Component {
       width,
     } = this.state;
 
+    let infraAffectedData = [];
+    const markDataPoints = 5;
+
+    changelogData.forEach((data) =>
+      scatterPlotData.some((dataPoint, index) => {
+        const affectedData = dataPoint.x > data.date;
```

Comment:
Rather than fetching all of the data and filtering it here by date, `startdate` and `enddate` filters should be added to the changelog API so we are only fetching data for the time period the user has set for the graph.  This would require a very simple change in `/webapp/api/changelog.py`. The thing to be careful about is that you're using a UTC timestamp.

Reply:
The endpoint isn't parameterized, yet it only sends data from the last 15 days. Note that Graphs view is configured to land on a data window of  14 days, which maps nicely to this setting.

Reply:
Will sync with the team to see whether it's indeed useful to fetch older changelogs (for say 30/60/90 days etc).